test/cases/union_depends_on_pointer_alignment.zig

```diff
@@ -0,0 +1,11 @@
+const U = union {
+    next: ?*align(1) U align(128),
+};
+
+export fn entry() usize {
+    return @alignOf(U);
+}
+
+// error
+//
+// :1:11: error: union layout depends on being pointer aligned
```

Comment:
Good case, but it should be in `test/cases/compile_errors/`.

Reply:
Moved it over.